frontend/scripts/react-components/shared/app.thunks.js

```diff
@@ -0,0 +1,67 @@
+import { SET_TOOLTIPS, SET_CONTEXTS, selectInitialContextById } from 'scripts/actions/app.actions';
+import {
+  GET_TOOLTIPS_URL,
+  getURLFromParams,
+  GET_CONTEXTS_URL
+} from 'scripts/utils/getURLFromParams';
+import { getCurrentContext } from 'scripts/reducers/helpers/contextHelper';
+
+function loadTooltipsPromise(dispatch, getState) {
+  const { app } = getState();
+  // we use forceReload to force state rehydration
+  if (app.tooltips !== null) {
+    return Promise.resolve();
+  }
+
+  const tooltipsURL = getURLFromParams(GET_TOOLTIPS_URL);
+
+  return new Promise(resolve =>
+    fetch(tooltipsURL)
+      .then(res => (res.ok ? res.json() : Promise.reject(res.statusText)))
+      .then(data => {
+        dispatch({
+          type: SET_TOOLTIPS,
+          payload: data
+        });
+      })
+      .then(() => resolve())
+  );
+}
+
+function loadContextsPromise(dispatch, getState) {
+  // Contexts should only load once
```

Comment:
I see that on initial load the context is being loaded twice, probably because of the language selecting redirect. Maybe we need some isLoading state?